flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -435,6 +435,16 @@ public class ExecutionConfigOptions {
                             "Determines whether CAST will operate following the legacy behaviour "
                                     + "or the new one that introduces various fixes and improvements.");
 
+    @Documentation.TableOption(execMode = Documentation.ExecMode.BATCH_STREAMING)
+    public static final ConfigOption<MapKeyDedupPolicy> TABLE_EXEC_MAPKEY_DEDUP_POLICY =
+            key("table.exec.mapkey-dedup-policy")
```

Comment:
I'm curious about why this option should be a global setting, how can user use different strategy for different function call of `map_from_entris` in a job ? Can it be an extra arguments of the function ?

Reply:
i do the surveys:
1) max compute supports  extra arguments , which user can use different strategy for different function call of map_from_entris in a job. https://www.alibabacloud.com/help/zh/maxcompute/latest/map-from-entries
2) while spark supports with global config https://github.com/apache/spark/blob/7e2c6c7ab23f75a6ba83baaa0545482a43845ce8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapBuilder.scala#L52
3) presto doesn't support the mapkey-dedup-policy, and spark supports this function later, so it suppots https://issues.apache.org/jira/browse/SPARK-23934
4) this function is not sql standard. and other colllection functions i supports following the spark way, so i does it too.

Reply:
And i have a questions, does it need to supports function level strategy
1) global setting can make it job level by offering this configure when using
2) function level strategy has 3 case
    1. all last win
    2. all throw exception
    3. some of last win and others throw exception. function level only make sense this case. but if one of the throw exception strategy has same key it also throws exption. the effective is same with all throw exception. So I think supporting function-level significance is not very meaningful. What do you think?